### ISSUE TITLE
`SLACK_ICON_EMOJI` not displaying correctly in some situations

### ISSUE THREAD

**Issue by MatthewCane** [rtCamp/action-slack-notify#179]
Firstly, thank you for this Action, I've been using this extensively and it's been awesome.

The issue I am facing is that the `SLACK_ICON_EMOJI` is working inconsistently for me. The relevant part of my workflow is:

```yaml
jobs:
  notify:
    name: Notify Slack
    runs-on: ubuntu-latest
    steps:
      - name: Slack Notification
        uses: rtCamp/action-slack-notify@v2
        env:
          SLACK_WEBHOOK: ${{ secrets.slack_webhook}}
          SLACK_USERNAME: Actions Bot
          SLACK_ICON_EMOJI: ${{ inputs.status == 'success' && ':rocket:' || ':siren:' }}
          SLACK_LINK_NAMES: true
          SLACK_COLOR: ${{ inputs.status }}
          SLACK_TITLE: ${{ inputs.title }}
          SLACK_MESSAGE: ${{ inputs.message }}
```

I am using the ternary operator as found in the [Actions documentation](https://docs.github.com/en/actions/learn-github-actions/expressions#example) to set the icon emoji depending on the `inputs.status` value which is either `success` or `failure`. When the status is `success` it displays the icon as  🚀 as expected, however when it is any other value it displays a generic icon.

I know the status is being passed correctly, because it has an effect on the `SLACK_COLOR` variable. I can also see it in the workflow log:

```
> Run rtCamp/action-slack-notify@v2
  env:
    SLACK_WEBHOOK: ***
    SLACK_USERNAME: Actions Bot
    SLACK_ICON_EMOJI: :siren:
    SLACK_LINK_NAMES: true
    SLACK_COLOR: failure
    SLACK_TITLE: Continuous Deployment Failure
    SLACK_MESSAGE: The CD process has failed. Please check the workflow for more information.
```

Is there anything I can do to resolve this? I can't understand why it isn't working as expected.

Thanks!

**Comment by L0RD-ZER0** [comment]
Tested with a bunch of emojis and random strings, and everything seems to work fine on our tests. Here, only problems I can think of are either `:siren:` emoji being used not present in whichever workspace you're trying to send the message to.

Even on a fresh slack workspace, there were no issues with any of the emotes present, but anything which wasn't present returned to a default image as it should. So maybe try out with a different emote?

Beyond that, I can't think of anything else what might be wrong here. Action ternary syntax seems right and can't figure out any problems. Leaving the issue open for now, feel free to reach out if you find any further issues, or to update on this one. Leaving the issue open for now, in case of any further developments.

**Comment by MatthewCane** [comment]
Thank you for the quick reply. After lots of testing I have found that a lot of emojis don't seem to work properly as icons in my Slack workspace. I am still not sure why this is the case, but I have found that 🆘 works fine so I have settled on using that.

For anyone that finds this thread with the same error, create a test webhook and use a test repo to try out lots of different emojis in a private channel.
